Assets/MRTK/Core/Definitions/SpatialAwareness/Experimental/SpatialAwarenessSceneObject.cs

```diff
@@ -10,34 +10,28 @@ namespace Microsoft.MixedReality.Toolkit.Experimental.SpatialAwareness
     public class SpatialAwarenessSceneObject : BaseSpatialAwarenessObject
     {
         public SpatialAwarenessSceneObject(
-            System.Guid guid,
+            int id,
```

Comment:
I might have missed this in the review.... where does this int id get generated? is it related in any way to the guid?

Reply:
After an offline discussion with Kurtis I decided to make guid internal to `WindowsSceneUnderstandingObserver` and use id in the events and other places that external code have access to (main reason being the int-based id is deeply coupled with several observer related scripts including interface(s)). The int id is generated on line 876 of the `WindowsSceneUnderstandingObserver` class by calling `GetHashCode()` on the guid. The observer also keeps a dictionary to enable lookups from int id to guid.